janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -419,6 +429,22 @@ public void setRetryOnConflict(Integer retryOnConflict) {
             this.retryOnConflict = retryOnConflict;
     }
 
+    public void setRetryAttemptLimit(int retryAttemptLimit) {
+        this.retryAttemptLimit = retryAttemptLimit;
+    }
+
+    public void setRetryInitialWaitMs(long retryInitialWaitMs) {
+        this.retryInitialWaitMs = retryInitialWaitMs;
+    }
+
+    public void setRetryMaxWaitMs(long retryMaxWaitMs) {
+        this.retryMaxWaitMs = retryMaxWaitMs;
+    }
+
+    public void setRetryOnErrorCodes(Set<Integer> retryOnErrorCodes) {
+        this.retryOnErrorCodes.addAll(retryOnErrorCodes);
+    }
```

Comment:
The method name is `setRetryOnErrorCodes`, but instead of `set` operation it `adds`.
You can fix it by actually setting `this.retryOnErrorCodes = retryOnErrorCodes` instead of mutating existing collection.
However, it seems there is no need to execute this method more than once during the lifetime of this class.
Thus, I would suggest moving all of these parameters into the constructor instead. I know that `setRetryOnConflict` and `setBulkRefresh` have the same problem, but I think it should be refactored.

Feel free to ignore this comment if you see any complications in this refactoring.

Reply:
Happy to make the change but for what it's worth that's going to turn the signature into:

```
public RestElasticSearchClient(RestClient delegate, int scrollKeepAlive, boolean useMappingTypesForES7,
                                   int retryAttemptLimit, Set<Integer> retryOnErrorCodes, long retryInitialWaitMs,
                                   long retryMaxWaitMs) {
```

Which may be pushing a parameter parameter count for transitioning to a builder. Especially if `setRetryOnConflict` and `setBulkRefresh` are added one day.

Reply:
@porunov Moved the new configuration settings into the constructor as suggested. Isolated that into commit c3961fc22. 

If that looks acceptable then I'll do the squash & sign-off to get the PR into a mergable state again. Figured I'd wait before squashing it in case you had second thoughts upon seeing it given my aforementioned comment.

Reply:
> Happy to make the change but for what it's worth that's going to turn the signature into:
> 
> ```
> public RestElasticSearchClient(RestClient delegate, int scrollKeepAlive, boolean useMappingTypesForES7,
>                                    int retryAttemptLimit, Set<Integer> retryOnErrorCodes, long retryInitialWaitMs,
>                                    long retryMaxWaitMs) {
> ```
> 
> Which may be pushing a parameter parameter count for transitioning to a builder. Especially if `setRetryOnConflict` and `setBulkRefresh` are added one day.

Yes, that's a refactoring issue, I agree. However, I think it still better to have a clear contract for this class creation. Later on this contract can be simplified by moving all these parameters to a separate "configuration" POJO for easier extensibility and clearer constructor.
I like it better how it looks right now than it was before. Thank you!